### PR TITLE
Syntax adjustments

### PR DESCRIPTION
**Pull request by JulianBissekkou**

### CODE REVIEW

File: example/injector_example.dart [hunk 1]
```diff
@@ -58,18 +59,18 @@ class TikkrDatabase extends Database {
 
 void main() {
   // Use this static instance
-  Injector injector = Injector.appInstance;
+  final injector = Injector.appInstance;
 
   //Register a dependency
```
Comment (saibotma) [inline example/injector_example.dart:64]:
```suggestion
  // Register a dependency
```

File: example/injector_example.dart [hunk 2]
```diff
@@ -58,18 +59,18 @@ class TikkrDatabase extends Database {
...
+  injector.registerDependency<Engine>(() => Engine());
 
-  injector.registerDependency<Car>((Injector injector) {
-    var engine = injector.getDependency<Engine>();
+  injector.registerDependency<Car>(() {
+    final engine = injector.getDependency<Engine>();
     return CarImpl(engine: engine);
   });
 
   //Maybe you want to register a class and you need it as a singleton
```
Comment (saibotma) [inline example/injector_example.dart:72]:
```suggestion
  // Maybe you want to register a class and you need it as a singleton
```

File: example/injector_example.dart [hunk 3]
```diff
@@ -58,18 +59,18 @@ class TikkrDatabase extends Database {
...
-  injector.registerSingleton<Database>((_) => TikkrDatabase());
+  injector.registerSingleton<Database>(() => TikkrDatabase());
 }
 
 //Now you can easily get your dependencies / singletons with one line
```
Comment (saibotma) [inline example/injector_example.dart:76]:
```suggestion
// Now you can easily get your dependencies / singletons with one line
```

File: lib/src/extension/compatibility_extension.dart [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import '../../injector.dart';
+
+/// Extensions that guarantee compatibility to with code written with
+/// previous version of the injector to ensure that the following updates
+/// don't break existing code.
+extension CompatibilityExtension on Injector {
```
Comment (saibotma) [inline lib/src/extension/compatibility_extension.dart:6]: That's a good Idea. 🚀

File: lib/src/extension/compatibility_extension.dart [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import '../../injector.dart';
+
+/// Extensions that guarantee compatibility to with code written with
+/// previous version of the injector to ensure that the following updates
```
Comment (saibotma) [inline lib/src/extension/compatibility_extension.dart:4]:
```suggestion
/// previous versions of Injector to ensure that the following updates
```

File: lib/src/extension/compatibility_extension.dart [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import '../../injector.dart';
+
+/// Extensions that guarantee compatibility to with code written with
```
Comment (saibotma) [inline lib/src/extension/compatibility_extension.dart:3]:
```suggestion
/// Extensions that guarantee compatibility to code written with
```

File: pubspec.yaml [hunk 7]
```diff
@@ -6,10 +6,11 @@ homepage: https://github.com/tappeddev/injector
 author: Tapped <contact@tapped.dev>
 
 environment:
-  sdk: '>=1.20.1 <3.0.0'
+  sdk: '>=2.7.2 <3.0.0'
```
Comment (saibotma) [inline pubspec.yaml:9]: Is `2.7.2` the minimum Version possible? If not I would try to lower it as much as possible.

File: pubspec.yaml [hunk 8]
```diff
@@ -6,10 +6,11 @@ homepage: https://github.com/tappeddev/injector
...
 
 dependencies:
-  meta: ^1.1.5
+  meta: ^1.1.8
 
 dev_dependencies:
-  test: ^1.5.2
\ No newline at end of file
+  test: ^1.15.1
+  extra_pedantic: 1.2.0
```
Comment (saibotma) [inline pubspec.yaml:16]: GitHub is telling me: No new line at the end of file ;)

File: analysis_options.yaml [hunk 9]
```diff
@@ -1,33 +1 @@
-analyzer:
```
Comment (stefanschaller) [inline analysis_options.yaml:1]: love it <3

File: lib/src/extension/register_extension.dart [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    String dependencyName = "",
+  }) =>
+      this.register(Factory.singleton(builder), override: override, dependencyName: dependencyName);
+
+  void registerDependency<T>(
+    Builder<T> builder, {
+    bool override = false,
+    String dependencyName = "",
+  }) =>
+      this.register(Factory.singleton(builder), override: override, dependencyName: dependencyName);
```
Comment (stefanschaller) [inline lib/src/extension/register_extension.dart:17]: should be ```Factory.factory ```. 

Let's write some tests. this could be a dangerous change

**Comment by stefanschaller** [comment]
We also need to update the readme

File: lib/src/extension/register_extension.dart [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import 'package:injector/injector.dart';
+import 'package:injector/src/factory/factory.dart';
+
+extension RegisterExtension on Injector {
```
Comment (stefanschaller) [inline lib/src/extension/register_extension.dart:4]: I don't understand why we need to create extensions for basic functionality.
I would prefer to add this directly to the injector

File: lib/src/injector_base.dart [hunk 12]
```diff
@@ -108,51 +59,56 @@ class Injector {
...
 
-    var instance = factory.instance as T;
+    final instance = factory.instance as T;
     _factoryCallIds.remove(factoryId);
 
     return instance;
   }
 
+  /// Shorter syntax for [get].
+  T call<T>({String dependencyName = ""}) => this.get<T>(dependencyName: dependencyName);
```
Comment (stefanschaller) [inline lib/src/injector_base.dart:97]: ```get``` is shorter than ```call``` :0

Comment (stefanschaller) [reply lib/src/injector_base.dart:97]: We did you added this method?

File: lib/src/injector_base.dart [hunk 13]
```diff
@@ -168,11 +124,12 @@ class Injector {
 
   /// Checks if [T] is actually set.
   void _checkValidation<T>() {
-    var type = T.toString();
+    final type = T.toString();
 
     if (T == dynamic) {
       throw Exception(
-          "No type specified !\nCan not register dependencies for type \"$type\"");
+        "No type specified !\nCan not register dependencies for type \"$type\"",
```
Comment (stefanschaller) [inline lib/src/injector_base.dart:131]: let's remove the trailing comma

File: pubspec.yaml (hunk 7 above)
Comment (stefanschaller) [reply pubspec.yaml:9]: True 🙈

**Comment by stefanschaller** [comment]
You removed the inner injector from the callback. How do you now handle stuff? Also some documentation and doc would be awesome

File: lib/src/injector_base.dart [hunk 14]
```diff
@@ -22,83 +22,34 @@ class Injector {
   /// The signature of a dependency consists of [T]
   /// and the optional [dependencyName].
   ///
-  /// The injector will always create new instances of dependencies that got
```
Comment (stefanschaller) [inline lib/src/injector_base.dart:25]: Why did you remove those examples / documentation?

File: lib/src/extension/register_extension.dart (hunk 11 above)
Comment (jogboms) [reply lib/src/extension/register_extension.dart:4]: I second this point of view

File: pubspec.yaml (hunk 7 above)
Comment (JulianBissekkou) [reply pubspec.yaml:9]: 2.6.0 works as well ✅

File: lib/src/extension/register_extension.dart (hunk 10 above)
Comment (JulianBissekkou) [reply lib/src/extension/register_extension.dart:17]: Nice catch! I also named the factory `provider` instead of `factory` because it was weird to have `Factory.factory`. I was looking for a more descriptive way to show that you create the instance overy time again 🤔

File: lib/src/extension/register_extension.dart (hunk 11 above)
Comment (JulianBissekkou) [reply lib/src/extension/register_extension.dart:4]: move it into the injector ✅

File: lib/src/injector_base.dart (hunk 12 above)
Comment (JulianBissekkou) [reply lib/src/injector_base.dart:104]: Call is the operator so we can use the injector similar to get it :

```dart

final car = injector.get<Car>();

final car2 = injector<Car>(); // <--- this is call
```

File: lib/src/injector_base.dart (hunk 13 above)
Comment (JulianBissekkou) [reply lib/src/injector_base.dart:138]: I kinda like this is that case since it doesn't break that weird. More personal pref tbh 🤷

**Comment by JulianBissekkou** [comment]
Updated the readme ✅ 
The innerInjector is not needed, not sure why we had it in the first place 🤔
